proactive-healthcheck/api-health-check.sh

```diff
@@ -0,0 +1,604 @@
+#!/bin/bash
+#set -x
+
+#input parameters
+#username to connect to the ProActive Portals
+username=changeme
+#password of the user connecting to the ProActive Portals
+proactiveAdminPasswordDecrypted=changeme
+#server DNS name or public IP
+serverDNSName=try.activeeon.com
+#server Port
+webPort=8443
+#web protocol, http or https
+webProtocol=https
+#name of one of the node sources, needed in a test to get its description
+nsName=local
+#timeout in seconds to check free nodes, job submission, and job status
+timeoutInSeconds=120
+
+#set colors for printf state of services
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+GREEN='\033[0;32m'
+PURPLE='\033[0;35m'
+
+#number of failed tests
+FAIL=0
+#number of successful tests
+SUCCESS=0
+
+#complete ProActive server web URL
+URL="${webProtocol}://${serverDNSName}:${webPort}"
+
+#tracks the number of seconds that have passed since the tests were started
+SECONDS=0
+
+#get the session id, needed for all tests
+endpoint=$(curl -o - -d "username=${username}&password=${proactiveAdminPasswordDecrypted}" --write-out "\n%{http_code}\n" --silent --output /dev/null ${URL}/rest/scheduler/login)
+response=($endpoint)
+sessionid=${response[0]}
+status=${response[1]}
+message="$NC: ${PURPLE}(POST: scheduler/login)$NC GET A SESSION ID"
+if [ "$status" == "200" ]; then
+  ((SUCCESS++))
+  echo -e "${GREEN} OK $message"
+else
+  echo -e "${RED} NOK $message"
+  ((FAIL++))
+  #exit the script if the session id could not be retrieved since it is required by most of the tests
+  exit 1
+fi
+
+#function used by tests to check the returned status and print messages
+check_endpoint() {
+  endpoint=$1
+  message=$2
+  if [[ "$endpoint" == "200" || "$endpoint" == "201" ]]; then
```

Comment:
as far as I know, if the response code starts with 2 it's ok it can 200 201 or other vaules
https://restfulapi.net/http-status-codes/

Reply:
well, I thought to apply 2xx pattern but as I can see in the doc 200 and 201 are the 100% correct ones. I don't want to generalize to avoid false positives. Then, all tests return 200 in any case